### DISCUSSION TITLE
Vercel is not able to read content of ```json``` file for my "GOOGLE_APPLICATION_CREDENTIALS" environment variable

### DISCUSSION THREAD

**Discussion by Veer0x1** [vercel/community#3203] [Help] [answered]
### Problem Description

I have json which contain service account credentials, I'm using it for [NextAuth Firebase Adapter](https://authjs.dev/reference/adapter/firebase) . I have this json in the root of Next.js project. When I'm runnig my application locally its working fine. But When I'm deploying it to vercel and adding this environment variable in my projects environment variable section I'm getting a error.
And just to run on it on production I'm also not gitignoring this json ( I know I should but don't know then how would I used it as ```env```)

Here is the screenshot of error.
<img width="1435" alt="Screenshot 2023-06-29 at 10 55 34 AM" src="https://github.com/vercel/vercel/assets/96714915/9f57dbd0-ae18-48ea-86c2-81a9e7cb20dc">

This is my environment variable.
<img width="907" alt="Screenshot 2023-06-29 at 10 56 35 AM" src="https://github.com/vercel/vercel/assets/96714915/8ac1ea8c-15f3-4aae-b38a-361c5c907eaf">

**Comment by amyegan** [comment]
Hi @Veer0x1! Sorry for the delay. I just noticed your question posted over in the CLI repo, so I moved it over here to our main discussion board. The right solution for you will depend on your app structure, the JSON data, etc. But I can think of a couple of options.

You could put the values from the JSON into their own individual environment variables. But if there are many values already in JSON format, I can see why you might prefer another option.

If the JSON file is small enough, you could stringify it and set it as an environment variable. Or you could use [Vercel CLI](https://vercel.com/docs/cli) to load the file contents into an env variable. The command would be something like `vercel env add FIREBASE_JSON production < ~/my-data.json`

**Reply by smaeda-ks** [reply, marked as answer]
@Veer0x1 just to understand your current solution, is the `GOOGLE_APPLICATION_CREDENTIALS` environment only to contain the file path, and not the JSON content itself?

If so, that can be problematic. To include custom files in your serverless functions, you may need to write your code bit more static way, such as: https://vercel.com/guides/how-can-i-use-files-in-serverless-functions
This is because we currently rely on https://github.com/vercel/nft to statically analyze your code to find out what files to be included. So if you use dynamic parameters in the file path in your code, that might not be analyzed correctly.

Another possible workaround is to use `experimental.outputFileTracingIncludes` option to explicitly specify the file:
https://nextjs.org/docs/pages/api-reference/next-config-js/output#caveats

**Reply by kpripper** [reply]
Isn't it logical to assume that the developer himself decides which files should be included in his application?
